Content.Server/Body/Systems/SmokeFilterSystem.cs

```diff
@@ -0,0 +1,119 @@
+using Content.Server.Atmos.Components;
+using Content.Shared.Clothing;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Inventory;
+using Content.Shared.FixedPoint;
+using Content.Shared.Examine;
+
+namespace Content.Server.Body.Systems;
+
+public sealed class SmokeFilterSystem : EntitySystem
+{
+    [Dependency] private readonly InventorySystem _inventory = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<FilterMaskComponent, ExaminedEvent>(OnMaskExamine);
+        SubscribeLocalEvent<FilterMaskComponent, GotEquippedEvent>(OnGotEquipped);
+        SubscribeLocalEvent<FilterMaskComponent, GotUnequippedEvent>(OnGotUnequipped);
```

Comment:
You may want to swap these for ClothingGot(Un)EquippedEvent, to make sure the effect is only active when the mask is worn in the right inventory slot (OnGotEquippedEvent is raised when items are put into pockets too).